src/parsers/trades/liquid/index.ts

```diff
@@ -0,0 +1,175 @@
+    currency_type: string;
+    direction: string;
+    transaction_id: string;
+    transaction_type: string;
+    gross_amount: string;
+    currency: string;
+    execution_id: string;
+    generated_for_type: string;
+    generated_for_id: string;
+    transaction_hash: string;
+    from_address: string;
+    to_address: string;
+    state: string;
+    created_at_jst: string;
+    updated_at_jst: string;
+    created_at_utc: string;
+    updated_at: string;
+    notes: string;
+    used_id: string;
+    account_id: string;
+}
+
+function groupByExecutionID(group: any, line: any) {
+    group[line.execution_id] = group[line.execution_id] ?? [];
+    group[line.execution_id].push(line);
+    return group;
+}
+
+export default async function processData(importDetails: IImport): Promise<ITrade[]> {
+    const data: ILiquid[] = await getCSVData(importDetails.data) as ILiquid[];
+    const internalFormat: ITrade[] = [];
+    const sorted = data.reduce(groupByExecutionID, {});
+    for (const execution of Object.keys(sorted)) {
+        const trades = sorted[execution];
+        const tradeToAdd: IPartialTrade = {
+            date : createDateAsUTC(new Date(trades[0].updated_at)).getTime(),
+            exchange : EXCHANGES.Liquid,
+            exchangeID : execution,
+        };
+        if (execution === '') {
+            for (const line of trades){
+                console.log(`Ignored ${tradeToAdd.exchange} trade of type ${line.transaction_type}`);
+                continue;
+            }
+        }
+        switch (trades[0].transaction_type) {
+            case 'trade': {
+                switch (trades.length) {
+                    case 1: {
+                        internalFormat.push(addSingleLineTrade(tradeToAdd, trades[0]));
+                        break;
+                    }
+                    case 2: {
+                        internalFormat.push(addTrade(tradeToAdd, trades[0], trades[1]));
+                        break;
+                    }
+                    case 3: {
+                        let feeTrade = tradeToAdd;
+                        internalFormat.push(addTrade(tradeToAdd, trades[0], trades[1]));
+                        internalFormat.push(addFeeTrade(feeTrade, trades[2]));
+                        break;
+                    }
+                    case 4: {
+                        let feeTrade = tradeToAdd;
+                        internalFormat.push(addTrade(tradeToAdd, trades[0], trades[1]));
+                        internalFormat.push(addFeeTrade(feeTrade, trades[2], trades[3]));
+                        break;
+                    }
+                    case 6: {
+                        internalFormat.push(addFeeTrade(tradeToAdd, trades[4], trades[5]));
+                        break;
+                    }
+                    case 8: {
+                        let secondTrade = tradeToAdd;
+                        internalFormat.push(addFeeTrade(tradeToAdd, trades[4], trades[5]));
```

Comment:
I dont really get this what happens to trades 0 - 3

Reply:
![image](https://user-images.githubusercontent.com/10415894/207631885-124def43-419a-4c5c-89e0-dd952b9ac73e.png)

Reply:
I don’t know why these trades over 6 or 8 lines exist but they all cancel themselves. Maybe a cancelled order or something

Reply:
I’ll send a request to Liquid support to check

Reply:
In the meantime, I changed the code so that even if it cancels itself, it shows up as a trade. So we actually import the data.